src/dolphin/atmosphere/troposphere.py

```diff
@@ -0,0 +1,425 @@
+            )
+    
+    return
+
+def prepare_geometry(geometry_dir: Path,
+                     geo_files: List[Path],
+                     matching_file: Path,
+                     dem_file: Optional[Path],
+                     epsg: int, out_bounds: Bbox,
+                     strides: dict[str, int] = {"x": 1, "y": 1}) ->dict[str, Path]: 
+    """
+    Prepare geometry files.
+
+    Parameters
+    ----------
+    geometry_dir : Path
+        Output directory for geometry files.
+    geo_files : List[Path]
+        List of geometry files.
+    matching_file : Path
+        Matching file.
+    dem_file : Optional[Path]
+        DEM file.
+    epsg : int
+        EPSG code.
+    out_bounds : Bbox
+        Output bounds.
+    strides : Dict[str, int], optional
+        Strides for resampling, by default {"x": 1, "y": 1}.
+
+    Returns
+    -------
+    Dict[str, Path]
+        Dictionary of prepared geometry files.
+    """
+    os.makedirs(geometry_dir, exist_ok=True)
+
+    stitched_geo_list = {}
+
+    # local_incidence_angle needed by anyone?
+    datasets = ["los_east", "los_north"]
+  
+    for ds_name in datasets:
+        outfile = geometry_dir / f"{ds_name}.tif"
+        print(f"Creating {outfile}")
+        stitched_geo_list[ds_name] = outfile
+        ds_path = f"/data/{ds_name}"
+        cur_files = [io.format_nc_filename(f, ds_name=ds_path) for f in geo_files]
+
+        no_data = 0
+
+        stitching.merge_images(
+            cur_files,
+            outfile=outfile,
+            driver="GTiff",
+            out_bounds=out_bounds,
+            out_bounds_epsg=epsg,
+            in_nodata=no_data,
+            out_nodata=no_data,
+            target_aligned_pixels=True,
+            strides=strides,
+            resample_alg="nearest",
+            overwrite=False,
+        )
+
+    
+    if dem_file:
+        height_file = geometry_dir / "height.tif"
+        stitched_geo_list['height'] = height_file
+        if not os.path.exists(height_file):
+            print(f"Creating {height_file}")
+            stitching.warp_to_match(
+                input_file=dem_file,
+                match_file=matching_file,
+                output_file=height_file,
+                resample_alg="cubic",
+            )
+
+    return stitched_geo_list
+
+
+def write_tropo(tropo_2d: dict, 
+                inp_interferogram: Path, 
+                out_dir: Path) -> None:
+    
+    """
+    Write tropospheric delay data to files.
+
+    Parameters
+    ----------
+    tropo_2d : Dict
+        Dictionary containing tropospheric delay data.
+    inp_interferogram : Path
+        Input interferogram file.
+    out_dir : Path
+        Output directory.
+    """
+    dates = os.path.basename(inp_interferogram).split('.')[0]
+    os.makedirs(out_dir, exist_ok=True)
+
+    for key, value in tropo_2d.items():
+        output = os.path.join(out_dir, f"{dates}_{key}.tif")
```

Comment:
```suggestion
        output = out_dir /  f"{dates}_{key}.tif"
```
(i think that's the last one of switching from `os.path` -> `pathlib.Path` functions)

Reply:
Thanks @scottstanie, I have changed as you suggested